db/sstables-format-selector.cc

```diff
@@ -53,8 +53,11 @@ future<> sstables_format_selector::maybe_select_format(sstables::sstable_version
 future<> sstables_format_selector::start() {
     assert(this_shard_id() == 0);
     co_await read_sstables_format();
-    _features.local().me_sstable.when_enabled(_me_feature_listener);
-    _features.local().md_sstable.when_enabled(_md_feature_listener);
+    // The listener may fire immediately, create a thread for that case.
+    co_await seastar::async([this] {
```

Comment:
This is a bug-fix, isn't it?

Reply:
Probably, but before my movements in `scylla_main` it didn't matter - features were loaded after
`sstables-format-selector::start`.

Reply:
This wrapping belongs more to when_enabled() impl.

Reply:
> This wrapping belongs more to when_enabled() impl.

Agree, but I already did a lot of refactorings in this PR, let's leave that for another.

Reply:
> let's leave that for another.

Ok

Reply:
We should make it explicit that listeners can yield (make them return a future). Threads are good in some local scope, but when gluing different components, we need to make the possibility of yielding clear.

Reply:
I [created an issue](https://github.com/scylladb/scylladb/issues/15347), will fix it in a follow-up.